neonvm/controllers/metrics.go

```diff
@@ -121,11 +154,22 @@ func (d *wrappedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	// async approach.
 	d.lock.Lock()
 	defer d.lock.Unlock()
+	outcome := SuccessOutcome
 	if err != nil {
 		d.failing[req.NamespacedName] = struct{}{}
+		if errors.IsConflict(err) {
+			outcome = ConflictOutcome
+		} else {
+			outcome = FailureOutcome
+		}
+
+		log.Error(err, "Failed to reconcile VirtualMachine",
+			"duration", duration.String(), "outcome", outcome)
 	} else {
 		delete(d.failing, req.NamespacedName)
+		log.Info("Successful reconciliation", "duration", duration.String())
```

Comment:
Would be good to check if Loki can filter on stringly-typed durations, or if it'd need to be a number. Otherwise, I'd suggest making this e.g. floating point number of seconds, rather than a pretty representation (unfortunately, because pretty representation would be much more readable)

Reply:
Yeah, it has e.g. `| duration > 10s`, and I think I used it already at some point.

Reply:
Ah awesome :)